### PR TITLE
test: fixed failing GitImport test

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
RCA:
Gitimport test was failing at the place of reconnect DS
And also later in 4th test during commit and push

Solution:
replaced some js helper with ts helper
updated commit and push to fix the test

Equivalent EE PR: https://github.com/appsmithorg/appsmith-ee/pull/4683

/ok-to-test tags="@tag.Sanity"

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9936172520>
> Commit: bb039490207a71fa4e411da1fb3ebe222c2ed2de
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9936172520&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Mon, 15 Jul 2024 09:03:21 UTC

## Summary by CodeRabbit

- **Tests**
  - Updated Git import flow test scenario to enhance control flow and method calls.
  - Improved commit and push command with additional assertions for element visibility and input in Git sync tests.
  - Updated spec name for limited tests in Cypress test configuration.

### CODE REVIEW

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/support/gitSync.js [hunk 1]
```diff
@@ -74,7 +74,15 @@ Cypress.Commands.add("switchGitBranch", (branch, expectError) => {
 Cypress.Commands.add("commitAndPush", (assertFailure) => {
   cy.get(homePage.publishButton).click();
   agHelper.AssertElementExist(gitSync._bottomBarPull);
-  cy.get(gitSyncLocators.commitCommentInput).type("Initial Commit");
+  agHelper.AssertElementVisibility(gitSyncLocators.gitSyncModal);
+  cy.get(gitSyncLocators.commitCommentInput).should(
+    "have.attr",
+    "placeholder",
+    "Your commit message here",
```
[the remark this answers is not in the document]
Comment (ApekshaBhosale) [reply app/client/cypress/support/gitSync.js:81]: btw why did you make change in this helper? @NandanAnantharamu

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/support/gitSync.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/support/gitSync.js:81]: it was waiting for the `gitSyncLocators.commitCommentInput` so added few steps to make sure it actually appears

File: app/client/cypress/support/gitSync.js [hunk 2]
```diff
@@ -74,7 +74,15 @@ Cypress.Commands.add("switchGitBranch", (branch, expectError) => {
 Cypress.Commands.add("commitAndPush", (assertFailure) => {
   cy.get(homePage.publishButton).click();
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/support/gitSync.js]: done

File: app/client/cypress/support/gitSync.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/support/gitSync.js]: done

File: app/client/cypress/limited-tests.txt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
+cypress/e2e/Regression/ClientSide/Git/GitImport/GitImport_spec.js
```
Comment (ApekshaBhosale) [inline app/client/cypress/limited-tests.txt:2]: revert this file please @NandanAnantharamu and run this on EE as well multiple times on last time

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**